src/viam/sdk/registry/registry.cpp

```diff
@@ -92,8 +155,61 @@ const google::protobuf::ServiceDescriptor* ResourceRegistration::service_descrip
     return service_descriptor_;
 }
 
+void register_resources() {
+    // Register all components
+    Registry::register_resource<BaseClient, BaseServer, component::base::v1::BaseService>(
```

Comment:
Yeah, I tried to figure out some way to do that with `boost::mpl::list` but got pretty lost in the weeds and walked away with the sense that `boost::mpl` wasn't able to do what I needed. I'm very very far from an expert on C++ meta-programming however, and more than happy to be pointed in a better direction or otherwise proven wrong!